firewood/src/merkle/proof.rs

```diff
@@ -473,41 +480,52 @@ impl<N: AsRef<[u8]> + Send> Proof<N> {
 
         match new_node.inner() {
             NodeType::Leaf(n) => {
-                let cur_key = n.path().0.as_ref();
+                let paths_match = n.path.iter().copied().eq(&mut key_nibbles);
 
-                if !key.contains_other(cur_key) {
```

Comment:
It looks like in the leaf node case, the old code only check if it contains the leaf's partial path, which probably is what we desired? Because to handle the non-existing proof case.

Reply:
I can't make it do the same thing as extension nodes. I actually think this code was incorrect as `key` could be `[1, 2, 3]` and `cur_key` could be `[2, 3]`. This case should early return but it wouldn't have with the old code, right?

Reply:
Not sure I understand this comment. `contains_other` is a horribly-named function that just checks that it starts with the same thing. These tests show that `[1,2,3]` and `[2,3]` do not return `true`.

```
    // false tests
    assert!(![1,2,3].as_slice().contains_other(&[2,3]));
    assert!(![2,3].as_slice().contains_other(&[1,2,3]));
    
    // true tests
    assert!([1,2,3].as_slice().contains_other(&[1,2]));
```

Reply:
Ah, "contains_other" implies a subset and was misleading, you are correct. In any case, the new code matches the old behaviour now